libs/primitives/src/lib.rs

```diff
@@ -336,3 +336,162 @@ pub mod liquidity_pools {
 		}
 	}
 }
+
+pub mod xcm {
+	use codec::{Compact, Encode};
+	use sp_io::hashing::blake2_256;
+	use sp_std::{borrow::Borrow, marker::PhantomData, vec::Vec};
+	use xcm::prelude::{
+		AccountId32, AccountKey20, Here, MultiLocation, PalletInstance, Parachain, X1,
+	};
+	use xcm_executor::traits::Convert;
+
+	/// NOTE: Copied from https://github.com/moonbeam-foundation/polkadot/blob/d83bb6cc7d7c93ead2fd3cafce0e268fd3f6b9bc/xcm/xcm-builder/src/location_conversion.rs#L25C1-L68C2
```

Comment:
I suppose clippy will complain about the url
```suggestion
	/// NOTE: Copied from <https://github.com/moonbeam-foundation/polkadot/blob/d83bb6cc7d7c93ead2fd3cafce0e268fd3f6b9bc/xcm/xcm-builder/src/location_conversion.rs#L25C1-L68C2>
```